source/unified-test-format/unified-test-format.rst

```diff
@@ -1377,6 +1379,19 @@ The structure of this object is as follows:
     <../server-discovery-and-monitoring/server-discovery-and-monitoring.rst#servertype>`__
     for a list of valid values.
 
+.. _expectedEvent_topologyDescriptionChangedEvent:
+
+- ``topologyDescriptionChangedEvent``: Optional object. Assertions for one or more
+  `TopologyDescriptionChangedEvent <../server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#events>`__ fields.
+
+  The structure of this object is as follows:
+
+  - ``previousDescription``: Optional object. A value corresponding to the topology
+    description as it was before the change that triggered this event.
+
+  - ``newDescription``: Optional object. A value corresponding to the topology
+    description as it was after the change that triggered this event.
```

Comment:
These fields are never used in the tests accompanying this change. Moreover, the spec doesn't address _how_ to compare these values. Contrast this with the existing rules for [serverDescriptionChangedEvent](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectedsdamevent), which defines a single `type` field that can be compared. There, `type` is also defined in the schema.

Using `object` types for these fields in the schema will make it difficult to distinguish tests that start asserting on these values, and without a schema bump we wouldn't necessarily know that new test runner rules were necessary. To that end, `object` should be used sparingly (we use that for operation arguments, but there is also guidance on raising errors for unsupported arguments).

With that in mind, I'd suggest removing the following from the schema:

```json
"properties": {
    "previousDescription": { "type": "object" },
    "newDescription": { "type": "object" }
}
```

When a test actually needs to start asserting on topology descriptions, we can start modeling fields like was done for ServerDescriptionChangedEvent.

It looks like you only need `topologyDescriptionChangedEvent` to exist for the purpose of [waitForEvent](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#waitforevent). If so, the prior art would be [expectedCmapEvents](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectedcmapevent), most of which are modeled as empty objects.

Reply:
Thanks, done.